### PR TITLE
Alerting: Refactor receiver_svc and provisioning config store into legacy_storage package

### PR DESCRIPTION
**Pull request by JacobsonMT**
**What is this feature?**

This PR contains some refactorings of `receiver_svc` and provisioning config store to facilitate easier reviews in upcoming work on the k8s receiver api implementation.

The changes contained here should be no-op functionality-wise, with the one exception that permission denied, and not found errors coming from the existing `receiver_svc` have changed to `errutils` so will be slightly different.

Changes:
- Adds tests for some missing coverage on the `api/v1/receivers` api.
- Creates a new package `legacy_storage` to hold the various k8s api `AlertmanagerConfig` deconstruction code and abstractions.
- Moves provisioning package config-related storage code (`alertmanagerConfigStoreImpl`) to `legacy_storage` for re-use. 

**Who is this feature for?**

Developers

### CODE REVIEW

File: pkg/services/ngalert/provisioning/contactpoints.go [hunk 1]
```diff
@@ -540,22 +538,10 @@ func RemoveSecretsForContactPoint(e *apimodels.EmbeddedContactPoint) (map[string
 	return s, nil
 }
 
-// handleWrappedError unwraps an error and wraps it with a new expected error type. If the error is not wrapped, it returns just the expected error.
-func handleWrappedError(err error, expected error) error {
```
Comment (JacobsonMT) [inline pkg/services/ngalert/provisioning/contactpoints.go:544]: NOTE: No longer necessary since we moved to errorutils in `receiver_scv` responses.

**Review by yuri-tceretian — APPROVED** [review]
LGTM

I think that the new receiver storage has too many responsibilities which taken from domain service. I understand why that was done but I think we could do it somehow differently (using a unit of work). Anyway, I think we can address that later because it will require to much of rework of the follow up PRs.

File: pkg/services/ngalert/models/receivers.go [hunk 2]
```diff
@@ -15,3 +22,39 @@ type GetReceiversQuery struct {
...
+
+func GetUIDFromNamed(t Named) string {
+	return GetUID(t.GetName())
+}
+
+func GetUID(name string) string {
+	sum := fnv.New64()
+	_, _ = sum.Write([]byte(name))
+	return fmt.Sprintf("%016x", sum.Sum64())
+}
```
[the remark this answers is not in the document]
Comment (JacobsonMT) [reply pkg/services/ngalert/models/receivers.go:61]: It is url safe yes, but base 16

**Comment by JacobsonMT** [comment]
> LGTM
> 
> I think that the new receiver storage has too many responsibilities which taken from domain service. I understand why that was done but I think we could do it somehow differently (using a unit of work). Anyway, I think we can address that later because it will require to much of rework of the follow up PRs.

Agreed, removed the `ReceiverStore` abstraction (and the Receiver model as it is no longer needed yet) from this PR in favour of some helper methods attached to `ConfigRevision`.

**Review by yuri-tceretian — APPROVED** [review]
LGTM! 🎉

**Pull request merged**